### PR TITLE
Handle vectorize indexing options

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Handle `$vectorize` field as special field for indexing options
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjector.java [hunk 1]
```diff
@@ -86,14 +86,21 @@ public static DocumentProjector createForIndexing(Set<String> allowed, Set<Strin
...
+      }
       return new DocumentProjector(ProjectionLayer.buildLayersOverlapOk(allowed), true, false);
     }
     if (denied != null && !denied.isEmpty()) {
       // (special) Case 4:
       if (denied.size() == 1 && denied.contains("*")) {
-        // Basically inclusion projector with nothing to include
+        // Basically inclusion projector with nothing to include but handle for $vector and
+        // $vectorize
+        Set<String> overrideFields = new HashSet<>();
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjector.java:99]: When used deny : ["*"], Need this logic. Fixed an existing bug.

**Review by tatu-at-datastax — APPROVED** [review]

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorizeSearchIntegrationTest.java [hunk 2]
```diff
@@ -475,6 +534,68 @@ public void happyPathWithInvalidData() {
...
+                  getAuthToken(),
+                  HttpConstants.EMBEDDING_AUTHENTICATION_TOKEN_HEADER_NAME,
+                  CustomITEmbeddingProvider.TEST_API_KEY))
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionNameDenyAll)
+          .then()
+          .statusCode(200)
+          .body("data.documents[0]._id", is("1"))
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorizeSearchIntegrationTest.java:563]: Maybe verify there's just one document returned?

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorizeSearchIntegrationTest.java [hunk 3]
```diff
@@ -475,6 +534,68 @@ public void happyPathWithInvalidData() {
...
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorizeSearchIntegrationTest.java:594]: Same as above, just one result doc check?

**Review by tatu-at-datastax — APPROVED** [review]

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Pull request merged**